src/main/java/com/hazelcast/aws/impl/DescribeInstances.java

```diff
@@ -72,10 +68,80 @@ public DescribeInstances(AwsConfig awsConfig, String endpoint) throws IOExceptio
         attributes.put("X-Amz-Expires", "30");
     }
 
+    DescribeInstances(AwsConfig awsConfig) {
+        this.awsConfig = awsConfig;
+    }
+
+    void checkKeysFromIamRoles(Environment env) throws IOException {
+        if (awsConfig.getAccessKey() != null) {
+            return;
+        }
+
+        if (awsConfig.getIamRole() == null) {

```

Comment:
non-existence of `iam-role` can be interpreted as default iam-role and if Amazon does not return valide keys for the default case, we anyhow throw `InvalidConfigurationException`. That means no configraution case will try to grap keys based on default iamrole.